web/admin/src/Admin.js

```diff
@@ -109,13 +115,33 @@ export default class Admin extends PureComponent {
     )
   }
 
+  saveCurrentIndex = (currentIndex) => {
+    this.setState({currentIndex})
+  }
+
   questionsForCurrentSession = () => Object.values(this.state.questionsBySession[this.state.sessionId] || {}).sort((a,b) => a.order - b.order)
 
   onSessionChange = e => this.setState({sessionId: e.target.value})
-  onSessionNameChange = e => this.sessionsRef().child(this.state.sessionId).update({name: e.target.value})
+
+  onSessionNameChange = e => {
+    const currentTitle = e.target.value.trim()
+    const dup = Object.values(this.state.sessions).find(i => i.name.toLowerCase() === currentTitle.toLowerCase())
+    if (currentTitle.length && !dup) {
+      this.publicSessionRef().child(this.state.sessionId).update({name: e.target.value, isDisplayable: true})
```

Comment:
It's better not to repeat these entire two lines for both cases.  The only thing different between lines 130/131 and 134/135 is the value of `isDisplayable`, so I would just have one copy of these two lines after assigning `isDisplayable` to `currentTitle.length > 0 && !dup`